<a href="https://colab.research.google.com/github/mariam7084/ML_projects/blob/main/Breast_Cancer_Detection_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###Importing the Libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


Dataset:

Wolberg,WIlliam. (1992). Breast Cancer Wisconsin (Original). UCI Machine Learning Repository. https://doi.org/10.24432/C5HP4Z.


In [ ]:
df = pd.read_csv('breast_cancer.csv')
x = df.iloc[:, 1:-1].values
y = df.iloc[:, -1].values

###Splitting the dataset into training set and test set

In [ ]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.25, random_state=0)

###Feature Scaling

In [ ]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
x_train = sc.fit_transform(x_train)
x_test = sc.transform(x_test)

#Logistic regression

In [ ]:
from sklearn.linear_model import LogisticRegression
classifier1 = LogisticRegression(random_state=42)
classifier1.fit(x_train, y_train)

LogisticRegression(random_state=42)

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier1.predict(x_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
accuracy_score(y_test, y_pred)

[[103   4]
 [  5  59]]


0.9473684210526315

##K-Nearest Neighbours

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
classifier2 = KNeighborsClassifier(n_neighbors = 5, metric = 'minkowski', p=2)
classifier2.fit(x_train, y_train)


KNeighborsClassifier()

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred2 = classifier2.predict(x_test)
cm = confusion_matrix(y_test, y_pred2)
print(cm)
accuracy_score(y_test, y_pred2)

[[103   4]
 [  5  59]]


0.9473684210526315

##Support Vector Machines

In [ ]:
from sklearn.svm import SVC
classifier3 = SVC(kernel = 'linear', random_state=42)
classifier3.fit(x_train, y_train)

SVC(kernel='linear', random_state=42)

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred3 = classifier3.predict(x_test)
cm = confusion_matrix(y_test, y_pred3)
print(cm)
accuracy_score(y_test, y_pred3)

[[102   5]
 [  5  59]]


0.9415204678362573

##Kernel SVM

In [ ]:
from sklearn.svm import SVC
classifier4 = SVC(kernel = 'rbf',random_state = 42)
classifier4.fit(x_train, y_train)

SVC(random_state=42)

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred4 = classifier4.predict(x_test)
cm = confusion_matrix(y_test, y_pred4)
print(cm)
accuracy_score(y_test,y_pred4)

[[101   6]
 [  3  61]]


0.9473684210526315

##Naive Bayes

In [ ]:
from sklearn.naive_bayes import GaussianNB
classifier5 = GaussianNB()
classifier5.fit(x_train, y_train)

GaussianNB()

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred5 = classifier5.predict(x_test)
cm = confusion_matrix(y_test, y_pred5)
print(cm)
accuracy_score(y_test, y_pred)

[[99  8]
 [ 2 62]]


0.9473684210526315

##Decison tree classification


In [ ]:
from sklearn.tree import DecisionTreeClassifier
classifier6 = DecisionTreeClassifier(criterion = 'entropy', random_state = 0)
classifier6.fit(x_train, y_train)


DecisionTreeClassifier(criterion='entropy', random_state=0)

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred6 = classifier6.predict(x_test)
cm = confusion_matrix(y_test, y_pred6)
print(cm)
accuracy_score(y_test, y_pred6)

[[104   3]
 [  4  60]]


0.9590643274853801

##Random Forest Classification

In [ ]:
from sklearn.ensemble import RandomForestClassifier
classifier7 = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 0)
classifier7.fit(x_train, y_train)

RandomForestClassifier(criterion='entropy', n_estimators=10, random_state=0)

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred7 = classifier7.predict(x_test)
cm = confusion_matrix(y_test, y_pred7)
print(cm)
accuracy_score(y_test, y_pred7)

[[104   3]
 [  5  59]]


0.9532163742690059

The highest accuracy score among the above 7 classification models was found to be 95.9% of the Decision Tree Classification Model.